### DISCUSSION TITLE
Jenkins Agent for CI/CD and connect to GitHub But can not run jar file in backround.

### DISCUSSION THREAD

**Discussion by minhchinhdo** [community/community#127919] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I use Jenkins Agent for CI/CD and connect to GitHub but in stage deploy everything is fine but when I run jar file in background nothing happens in jenkins pipeline but in server it is fine

**Comment by Abhishek22K** [comment]
It might be related to how Jenkins background processes or jobs in your pipeline. You can add logging statements in your Jenkins pipeline script or instead of running the jar file in the background try running it in the foreground.

**Reply by minhchinhdo** [reply]
The process is I build and deploy in once server (vps), also config jenkins agent, outbound in this server. Maybe, I think the problem has become here when I trying to run jar file in backround: 

After build and clone code to dir /var/lib/jenkins -> Copy jar file to a folder using to deploy and I try run jar file then > nohup.out 2>&1 & in this statement order to run in backround.

**Reply by minhchinhdo** [reply, marked as answer]
Update: deployment in backround is fine if i run it on server but in jenkins nothing happens.

**Comment by Narendrakaduru** [comment]
Just place the jar out of ${WORKSPACE}

environment {
        APP_NAME = "hello-java-app"
        JAR_NAME = "hello-java-app.jar"
        BASE_DIR = "${WORKSPACE}"
        STAGING_DIR = "/home/user/${APP_NAME}/staging"
        RELEASE_DIR = "/home/user/${APP_NAME}/release"
        BACKUP_DIR  = "/home/user/${APP_NAME}/backup"
        SOURCE_JAR = "target/hello-java-app.jar"
    }

 run the jar using different user
 echo "================  Start application ================"
  sudo chmod 777 ${RELEASE_DIR}/${JAR_NAME}
  sudo -u user bash -c "setsid java -jar ${RELEASE_DIR}/${JAR_NAME} &"
